src/requests.rs

```diff
@@ -0,0 +1,117 @@
+use crate::auth::{get_api_app_context, get_api_user_context};
+use std::collections::HashSet;
+use time::{Date, OffsetDateTime};
+use twitter_v2::id::NumericId;
+use twitter_v2::query::Exclude::Replies;
+use twitter_v2::query::TweetField::{AuthorId, CreatedAt, InReplyToUserId};
+
+const MINIMUM_NUMBER_OF_RESULTS: usize = 5;
+const MAXIMUM_NUMBER_OF_RESULTS: usize = 100;
+const DEFAULT_BADGE_SIZE: usize = 10;
+
+// Gets id of currently authorized user.
+#[allow(dead_code)]
+pub async fn get_my_user_id() -> Option<NumericId> {
+    let api = get_api_user_context();
+    let me = api
+        .get_users_me()
+        .send()
+        .await
+        .expect("invalid user")
+        .into_data();
+
+    match me {
+        None => None,
+        Some(me) => Option::from(me.id),
+    }
+}
+
+// Gets id of the latest reply of given user.
+#[allow(dead_code)]
+pub async fn get_latest_reply_id(user: NumericId) -> Option<NumericId> {
+    let api = get_api_app_context();
+    let my_tweets = api
+        .get_user_tweets(user)
+        .tweet_fields([InReplyToUserId])
+        .max_results(MINIMUM_NUMBER_OF_RESULTS)
+        .send()
+        .await
+        .expect("invalid user")
+        .into_data();
+
+    match my_tweets {
+        None => None,
+        Some(tweets) => {
+            for tweet in tweets {
+                if tweet.in_reply_to_user_id.is_some() {
+                    return Option::from(tweet.id);
+                }
+            }
+
+            None
+        }
+    }
```

Comment:
```suggestion
    my_tweets?.iter().find(|tweet| tweet.in_reply_to_user_id.is_some()).map(|tweet| tweet.id) 
```
takie przepisywanie biblioteki standardowej trochę pokazuje brak obycia z nią :/

Reply:
Przepraszam, poprawione.